processing/src/main/java/org/apache/carbondata/processing/loading/parser/impl/RowParserImpl.java

```diff
@@ -69,7 +69,12 @@ public RowParserImpl(DataField[] output, CarbonDataLoadConfiguration configurati
     DataField[] input = new DataField[fields.length];
     inputMapping = new int[input.length];
     int k = 0;
+    Boolean isNonSchemaPresent = false;
     for (int i = 0; i < fields.length; i++) {
+      if (fields[i].getColumn().getSchemaOrdinal() == -1) {
+        isNonSchemaPresent = true;
+        continue;
+      }
```

Comment:
if (fields[i].getColumn().getSchemaOrdinal() == -1) {
        input[k] = fields[i];
        inputMapping[k] = -1;
        k++;
        continue;
      }

Reply:
This could have been the easiest thing to do for me :)
There was a reason to keep inputMapping for non schema columns to be at the end. We can stop the row parse moment a non schema columns is encountered. We have only 1 non schema columns at the moment though. 

Anyway, have modified as suggested.